File: LeetCode/2500-2600/2540. Minimum Common Value/Solution.java
// Two Pointers method

class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        int ptr1 = 0, ptr2 = 0;
        int m = nums1.length, n = nums2.length;

        while (m > ptr1 && n > ptr2) {
            if (nums1[ptr1] == nums2[ptr2])
                return nums1[ptr1];
            else if (nums1[ptr1] > nums2[ptr2])
                ptr2++;
            else
                ptr1++;
        }
        return -1;
    }
}

File: LeetCode/2500-2600/2540. Minimum Common Value/Solution.py
# Binary Search method
# TC: O(n * logm)
# SC: O(1)

class Solution(object):
    def getCommon(self, nums1, nums2):
        m, n = len(nums1), len(nums2)
        for key in nums1:
            left, right = 0, m - 1
            while right >= left:
                mid = (left + right) // 2
                if nums2[mid] == key: return key
                elif nums2[mid] > key: right = mid - 1
                else: left = mid + 1
        return -1